File: backend/voting-app/src/main/java/com/tu/votingapp/entities/surveys/SurveyOptionEntity.java
package com.tu.votingapp.entities.surveys;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "survey_options")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class SurveyOptionEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "option_text")
    private String optionText;

    @Column(name = "vote_count")
    private int voteCount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id")
    private SurveyQuestionsEntity question;
}

File: frontend/android/app/src/main/java/com/example/android/entity/survey/SurveyOptionEntity.kt
package com.example.android.entity.survey

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "survey_options")
data class SurveyOptionEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id") val id: Long? = null,

    @ColumnInfo(name = "option_text") val optionText: String,
    @ColumnInfo(name = "vote_count") val voteCount: Int = 0,
    @ColumnInfo(name = "question_id") val questionId: Long
)